pkg/server/server_test.go

```diff
@@ -231,16 +233,260 @@ func TestServerWithMySQLDatastoreAndExplicitCredentials(t *testing.T) {
+		StoreId: storeID,
+		Writes: &openfgav1.WriteRequestWrites{
+			TupleKeys: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "document:1#viewer"),
+				tuple.NewTupleKey("document:1", "editor", "document:1#viewer"),
+				tuple.NewTupleKey("document:1", "editor", "user:andres"),
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	var wg sync.WaitGroup
+
+	wg.Add(2)
+
+	var resp1 *openfgav1.CheckResponse
+	var err1 error
+	go func() {
+		defer wg.Done()
+
+		resp1, err1 = s.Check(context.Background(), &openfgav1.CheckRequest{
+			StoreId:              storeID,
+			AuthorizationModelId: modelID,
+			TupleKey:             tuple.NewCheckRequestTupleKey("document:1", "editor", "user:jon"),
+		})
+	}()
+
+	var resp2 *openfgav1.CheckResponse
+	var err2 error
+	go func() {
+		defer wg.Done()
+
+		resp2, err2 = s.Check(context.Background(), &openfgav1.CheckRequest{
+			StoreId:              storeID,
+			AuthorizationModelId: modelID,
+			TupleKey:             tuple.NewCheckRequestTupleKey("document:1", "viewer", "user:jon"),
+		})
+	}()
+
+	wg.Wait()
+
+	require.NoError(t, err1)
+	require.False(t, resp1.GetAllowed())
+
+	require.NoError(t, err2)
+	require.False(t, resp2.GetAllowed())
+}
+
+func TestBlerbAvoidDeadlockWithinRequest(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+
+	_, ds, _ := util.MustBootstrapDatastore(t, "memory")
+
+	s := MustNewServerWithOpts(
+		WithDatastore(ds),
+	)
+	t.Cleanup(s.Close)
+
+	createStoreResp, err := s.CreateStore(context.Background(), &openfgav1.CreateStoreRequest{
+		Name: "openfga-test",
+	})
+	require.NoError(t, err)
+
+	storeID := createStoreResp.GetId()
+
+	model := testutils.MustTransformDSLToProtoWithID(`model
+			schema 1.1
+
+		  type user
+
+		  type document
+			relations
+			  define editor1: [user, document#viewer1]
+
+			  define viewer2: [document#viewer1] or editor1
+			  define viewer1: [user] or viewer2
+			  define can_view: viewer1 or editor1`)
+
+	writeAuthModelResp, err := s.WriteAuthorizationModel(context.Background(), &openfgav1.WriteAuthorizationModelRequest{
+		StoreId:         storeID,
+		SchemaVersion:   model.GetSchemaVersion(),
+		TypeDefinitions: model.GetTypeDefinitions(),
+	})
+	require.NoError(t, err)
+
+	modelID := writeAuthModelResp.GetAuthorizationModelId()
+
+	_, err = s.Write(context.Background(), &openfgav1.WriteRequest{
+		StoreId: storeID,
+		Writes: &openfgav1.WriteRequestWrites{
+			TupleKeys: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "editor1", "document:1#viewer1"),
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	resp, err := s.Check(context.Background(), &openfgav1.CheckRequest{
+		StoreId:              storeID,
+		AuthorizationModelId: modelID,
+		TupleKey:             tuple.NewCheckRequestTupleKey("document:1", "can_view", "user:jon"),
+	})
+	require.NoError(t, err)
+	require.False(t, resp.GetAllowed())
+}
+
+func TestBlerbThreeProngThroughVariousLayers(t *testing.T) {
```

Comment:
```suggestion
func TestThreeProngThroughVariousLayers(t *testing.T) {
```

I'm sure there's a more descriptive name for this but I can't think of one. Anything besides blerb tho lol.